### DISCUSSION TITLE
Trigger a RunRequest on GitHub repository actions

### DISCUSSION THREAD

**Discussion by randerzander** [dagster-io/dagster#7312] [Q&A]
Hello,

I'm getting started on a POC with Dagster. I'd like to trigger job executions on events happening in a remote GitHub repository:

For example:

1. An issue being opened
2. A PR being opened
3. A PR being merged

I see the [dagster-github](https://docs.dagster.io/_apidocs/libraries/dagster-github) integration docs, and browsed the source, but it didn't seem like there's a way to do this.

Seems like my best bet might be to have a custom Sensor use [PyGitHub](https://github.com/PyGithub/PyGithub) or shell out to the GitHub CLI and generate RunRequests that way?

Thanks!

**Comment by yuhan** [comment]
hi @randerzander I think there are a few options:
1. Let Dagster fully manage the flow end-to-end: write a custom sensor that listens to GitHub events and yield a RunRequest on certain conditions
2. Use GitHub's built-in event triggering scheme (e.g. GitHub actions or webhooks )and launch a run using one of Dagster Python API, GraphQL API, or CLI.
    * For example, we use github action to manage issues: [code](https://github.com/dagster-io/dagster/blob/master/.github/workflows/oss-issues.yml#L4-L7). Similarly, when an issue is opened, you can set it up in the `jobs` so it runs a Dagster CLI to launch a run.
    * Here's an example to launch a run via GraphQL: https://docs.dagster.io/concepts/dagit/graphql#launch-a-run
